lib/ddtrace/tracer.rb

```diff
@@ -212,8 +212,8 @@ def start_span(name, options = {})
         # child span
         span.parent = parent # sets service, trace_id, parent_id, sampled
       end
-      @tags.each { |k, v| span.set_tag(k, v) } unless @tags.empty?
-      tags.each { |k, v| span.set_tag(k, v) } unless tags.empty?
+      span.set_tags(@tags) unless @tags.empty?
+      span.set_tags(tags) unless tags.empty?
```

Comment:
wonder if the `unless tags.emoty` is even necessary (I mean it is not, but from readability point of view?)

Reply:
Doesn't look necessary strictly speaking. The question is what's less expensive; checking `empty?` or entering another function, then `each` on an empty hash?

It's probably less expensive doing `empty?` if I had to guess; you could benchmark it if you were so inclined, but I'd be fine with it as it is.